src/client/meshgen/item.cpp

```diff
@@ -0,0 +1,233 @@
+#include "item.h"
+#include "client.h"
+#include "collector.h"
+#include "content_mapblock.h"
+#include "itemdef.h"
+#include "mesh.h"
+#include "nodedef.h"
+
+class ItemMeshGenerator
+{
+	Client *client = nullptr;
+	ITextureSource *tsrc = nullptr;
+	const ItemDefinition *def = nullptr;
+	const ContentFeatures *f = nullptr;
+	content_t id = CONTENT_IGNORE;
+	bool is_flat = true;
+
+	scene::SMesh *mesh = nullptr;
+	std::vector<ItemPartColor> colors;
+
+	bool tryImage(const std::string &image, const std::string &overlay, bool extrude);
+	bool tryNode();
+	void createSpecialNodeMesh();
+	void postProcessMesh();
+	void postProcessNodeMesh();
+
+public:
```

Comment:
Could you please move these public functions to the top of the class definition? Many (/all?) classes are formatted that way.